### ISSUE TITLE
Need Help With Example

### ISSUE THREAD

**Issue by jmwright** [adam-james-v/svg-clj#7]
I tried out [svg-cli-interactive](https://adam-james-v.github.io/svg-clj-interactive/index.html) and attempted to paste in the example blossom from the readme. The only change I made was to replace `(tools/save-svg blossom "examples/blossom.svg")` with `blossom` to render the object. The web UI started to lag pretty badly and the controls do not seem to update the SVG. The SVG that is rendered also does not look like it's correct to me. I don't see any errors in the web console This is in Firefox 91.0.2 on Ubuntu 18.04 64-bit.

![Screenshot from 2021-08-30 14-33-26](https://user-images.githubusercontent.com/1015439/131387955-b975bf7d-79f6-4052-bcc3-56db35387f4b.png)

**Comment by adam-james-v** [comment]
Thanks for reporting this. I think I know what is causing the graphical errors. I think it’s related to a bug in my path namespace. When a float value is written in scientific notation, the lowercase e breaks an assumption I made about path string splitting. 

If you change the rotation value in petal to 90.001 or similar, I think it’ll fix the render. 

I will push an update to the webpage fixing this bug tomorrow. 

As for the lag, I am aware that I need to improve efficiency with the library, but don’t have a clear path forward yet. I could turn off automatic evaluation which would keep the page snappy, but I’m not sure if that’s nice for usability.

**Comment by jmwright** [comment]
> If you change the rotation value in petal to 90.001 or similar, I think it’ll fix the render.

So do I change all the `tf/rotate` entries with (/ x y) statements to `90.001`? Sorry, I'm not up to speed on the code.

```clojure
(tf/rotate 90.001)
```

>  I could turn off automatic evaluation which would keep the page snappy, but I’m not sure if that’s nice for usability.

Could you add a delay after the user stops typing to start the evaluation? That was the hardest part for me. I started typing "blossom" and it would re-render after each keypress.

**Comment by adam-james-v** [comment]
I think you only have to change the petal function:

```clojure 
(defn petal
  [cpts]
  (let [beza (apply path/bezier cpts)
        bezb (apply path/bezier (flip-y cpts))
        shape (tf/merge-paths beza bezb)
        ctr (tf/centroid shape)]
    (-> shape
        (tf/rotate -90.001) ;;add or remove decimal places to see if things work
        (tf/translate (utils/v* ctr [-1 -1])))))
```

I like the idea of adding a delay to the eval, I’ll work on that!

**Comment by jmwright** [comment]
> I think you only have to change the petal function:

Ah, ok. I can confirm that it does fix the render.

**Comment by adam-james-v** [comment]
I have addressed the rendering issue in a snapshot push earlier, and re-compiled the interactive site with the new code.

I have also attempted to help the lag issue by debouncing the eval while the user types. There is a 2 second eval delay which helps a bit, though I would like to solve this problem better. I think using a diffing strategy is best, but I don't know exactly how to implement that yet.

[This push](https://github.com/adam-james-v/svg-clj-interactive/commit/a7c9562d57f01e2f5fca8b575c07938edaeb2e15) incorporates the above changes to the interactive site.

I'm opening an issue for this UI lag on the svg-clj-interactive project and closing this issue.

[Here's the new issue.](https://github.com/adam-james-v/svg-clj-interactive/issues/1)

Thanks @jmwright for posting this issue.
